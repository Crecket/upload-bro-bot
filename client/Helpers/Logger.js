const log = require("loglevel");

// set logging level based on env
log.setLevel(process.env.DEBUG === "true" ? "trace" : "warn");

// export the logger
module.exports = log;
