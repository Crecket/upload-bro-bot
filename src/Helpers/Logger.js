const pino = require('pino');
const chalk = require('chalk');

// disable/enable based on enviroment and debug settings
const DEV = process.env.DEBUG || process.env.NODE_ENV !== "production";

// value helpers for pretty print
const levelColors = {
    default: chalk.white,
    60: chalk.bgRed,
    50: chalk.red,
    40: chalk.yellow,
    30: chalk.green,
    20: chalk.blue,
    10: chalk.grey
};
const levelLabels = {
    default: 'USERLVL',
    60: 'FATAL',
    50: 'ERROR',
    40: 'WARN',
    30: 'INFO',
    20: 'DEBUG',
    10: 'TRACE'
}

// export the logger
module.exports = pino({
    prettyPrint: DEV ? {
        levelFirst: true,
        formatter: (msg) => {
            // get chalk output bassed on level
            const outputLabel = levelColors[msg.level](levelLabels[msg.level]);
            // combine and return
            return `${outputLabel} ${msg.msg}`;
        }
    } : false,
    level: DEV ? "trace" : "info"
});

// module.exports = {
//     log: console.log,
//     error: console.error,
//     warn: console.log,
//     fatal: console.error,
//     debug: console.log,
//     info: console.log,
// };