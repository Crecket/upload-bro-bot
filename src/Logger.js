module.exports = {
    debug: process.env.DEBUG === "true" ? console.log : () => {
        },
    log: console.log,
    warn: console.log,
    error: console.error,
};