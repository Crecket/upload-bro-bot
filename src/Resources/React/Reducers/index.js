import {combineReducers} from "redux";

import modal from "./modalReducers";
import user from "./user";

export default combineReducers({
    modal,
    user,
})
