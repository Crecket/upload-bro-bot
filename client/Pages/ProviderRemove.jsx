import React from "react";
import Helmet from "react-helmet";
import Paper from "material-ui/Paper";
import CircularProgress from 'material-ui/CircularProgress';
import RaisedButton from 'material-ui/RaisedButton';
import Check from 'material-ui/svg-icons/navigation/check';
import Error from 'material-ui/svg-icons/alert/error';
import {red500, red800, green800} from 'material-ui/styles/colors';

import Utils from '../Helpers/Utils';
import Logger from '../Helpers/Logger';
import NavLink from '../Components/Sub/NavLink.jsx';
import axios from 'axios';

const styles = {
    img: {
        height: 120,
        padding: 20
    },
    button: {
        width: '100%'
    },
    inputs: {
        width: '100%',
    },
    paperWrapper: {
        marginTop: 40,
        textAlign: 'center'
    },
    paper: {
        width: '100%',
        padding: 20,
        textAlign: 'center',
        display: 'inline-block',
    },
    checkIcon: {
        width: 60,
        height: 60,
        color: green800
    },
    errorIcon: {
        width: 60,
        height: 60,
        color: red500
    }
};

class ProviderRemove extends React.Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            loadingState: "nope",
            error: false
        };
    }

    componentWillMount() {
        // check if provider is available
        if (!this.props.sites[this.props.params.type]) {
            this.props.router.push('/dashboard');
        }
    }

    removeProvider = () => {
        let providerType = this.props.params.type;

        // send our hashtag data
        axios.post("/api/remove/" + providerType, {_csrf: csrfToken})
            .then((result) => {
                if (result.data) {
                    // update state
                    this.setState({loadingState: "removed"});

                    // update the current user
                    this.props.updateUser();
                    setTimeout(() => {
                        this.props.router.push('/dashboard');
                    }, 2000);
                } else {
                    this.setState({error: true});
                    setTimeout(() => {
                        this.props.router.push('/dashboard');
                    }, 2000);
                }
            })
            .catch((error) => {
                Logger.error(error);
                this.setState({error: true});
                setTimeout(() => {
                    this.props.router.push('/dashboard');
                }, 2000);
            })
    }

    render() {
        // check if provider is available
        if (!this.props.sites[this.props.params.type]) {
            return null;
        }

        // extract name
        let providerType = Utils.ucfirst(this.props.params.type);

        let removeDiv = (
            <div className="row around-xs">
                <div className="col-xs-12">
                    <h1>Remove {providerType}?</h1>
                    <p>This will disable {providerType} service for UploadBro</p>
                </div>

                <div className="col-xs-12">
                    <img src={this.props.sites[this.props.params.type].logos['svg']}
                         style={styles.img}/>
                </div>

                <div className="col-xs-6 col-md-4">
                    <NavLink to="/dashboard">
                        <RaisedButton primary={true}>
                            Cancel
                        </RaisedButton>
                    </NavLink>
                </div>

                <div className="col-xs-6 col-md-4">
                    <RaisedButton secondary={true} backgroundColor={red800}
                                  onClick={this.removeProvider}>
                        Remove
                    </RaisedButton>
                </div>
            </div>
        )
        if (this.state.error) {
            removeDiv = (
                <h3>
                    <Error style={styles.errorIcon}/>
                    <br/>
                    Something went wrong
                </h3>
            )
        } else if (this.state.loadingState === "loading") {
            removeDiv = (
                <div className="row around-xs">
                    <div className="col-xs-12">
                        <h1>Removing {providerType}</h1>
                    </div>

                    <div className="col-xs-12">
                        <div className="box">
                            <CircularProgress color="rgb(28, 142, 215)" size={80} thickness={6}/>
                        </div>
                    </div>
                </div>
            )
        } else if (this.state.loadingState === "removed") {
            removeDiv = (
                <h3>
                    <Check style={styles.checkIcon}/>
                    <br/>
                    Removed your {providerType} account
                </h3>
            )
        }

        return (
            <div style={styles.paperWrapper}
                 className="col-xs-12 col-md-8 col-md-offset-2 col-lg-6 col-lg-offset-3">
                <Paper style={styles.paper}>
                    <Helmet>
                        <title>{`UploadBroBot - Remove ${providerType}`}</title>
                    </Helmet>
                    {removeDiv}
                </Paper>
            </div>
        );
    }
}

export default ProviderRemove;
