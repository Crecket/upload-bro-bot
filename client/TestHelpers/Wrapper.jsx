"use strict";
import React from "react";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import Dark from "../Themes/Dark.jsx";

// navigator fallback for server-side rendering
const navigatorHelper =
    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36";

class Wrapper extends React.PureComponent {
    constructor(props, context) {
        super(props, context);
        this.state = {};
    }

    render() {
        // wrap the children in the muitheme for jest tests
        return (
            <MuiThemeProvider
                muiTheme={getMuiTheme(Dark, { userAgent: navigatorHelper })}
                {...this.props}
            >
                {this.props.children}
            </MuiThemeProvider>
        );
    }
}

export default Wrapper;
