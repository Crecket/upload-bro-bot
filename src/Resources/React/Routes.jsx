import React from 'react';
import {Route, IndexRoute} from 'react-router';

import Main from './Components/Main';
import Home from './Pages/Home';
import Login from './Pages/Login';
import NotFound from './Pages/NotFound';
import DropboxLoginCallback from './Pages/DropboxLoginCallback';
import GoogleLoginCallback from './Pages/GoogleLoginCallback';

export default (
    <Route path="/" component={Main}>
        {/* default route */}
        <IndexRoute component={Home}/>

        {/* Routes*/}
        <Route path='/login' component={Login}/>
        <Route path='/login/google/callback' component={GoogleLoginCallback}/>
        <Route path='/login/dropbox/callback' component={DropboxLoginCallback}/>

        {/* fall back route, if no others are found show 404*/}
        <Route path='*' component={NotFound}/>
    </Route>
);