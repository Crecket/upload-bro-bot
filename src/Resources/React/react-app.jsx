import React from 'react';
import ReactDOM  from 'react-dom';
import {Provider} from "react-redux";
import injectTapEventPlugin  from 'react-tap-event-plugin';
import Logger from './Helpers/Logger.js';

// main app
import Main from './Components/Main';
import store from "./store";

// injection, required for materialze tap events
injectTapEventPlugin();

// render the react app
ReactDOM.render(
    <Provider store={store}>
        <Main />
    </Provider>,
    document.getElementById('app')
);

Logger.debug('Mounted react succesfully');
