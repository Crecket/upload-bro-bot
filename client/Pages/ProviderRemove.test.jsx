"use strict";

import React from "react";
import { StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import moxios from "moxios";

import Wrapper from "../TestHelpers/Wrapper.jsx";
import Logger from "../Helpers/Logger";
import ProviderRemove from "./ProviderRemove.jsx";

// pre-loaded info from json
const userInfoList = require("../TestHelpers/Data/api-get_user.json");
const siteInfoList = require("../TestHelpers/Data/api-get_providers.json");
const siteBoxInfo = require("../TestHelpers/Data/api-get_provider_box.json");
// const siteGoogleInfo = require('../TestHelpers/Data/api-get_provider_google.json');

const defaultTestProps = {
    match: {
        params: { type: "google" }
    },
    sites: siteInfoList,
    user_info: userInfoList
};

describe("<ProviderRemove />", () => {
    beforeEach(function() {
        // import and pass your custom axios instance to this method
        moxios.install();
    });

    afterEach(function() {
        // import and pass your custom axios instance to this method
        moxios.uninstall();
    });

    it("matches snapshot", () => {
        //create component and save as json
        const component = renderer.create(
            <StaticRouter>
                <Wrapper>
                    <ProviderRemove {...defaultTestProps} />
                </Wrapper>
            </StaticRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("123 short swipe to right event does not trigger router push event", () => {
        // mock router push function
        const routerPush = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove)

        // trigger on swipe move/end event
        ProviderRemoveWrapper.simulate('mousemove',{
            pageX: 1000
        });
        ProviderRemoveWrapper.simulate('mouseup');
        // ProviderRemoveWrapper.onSwipeMove({ x: 10 });
        // ProviderRemoveWrapper.onSwipeEnd();

        // router push should've bene called
        expect(routerPush).toHaveBeenCalledTimes(0);
    });

    it("short swipe to left event does not trigger router push event", () => {
        // mock router push function
        const routerPush = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove).instance();

        // trigger on swipe move/end event
        ProviderRemoveWrapper.onSwipeMove({ x: -10 });
        ProviderRemoveWrapper.onSwipeEnd();

        // router push should've bene called
        expect(routerPush).toHaveBeenCalledTimes(0);
    });

    it("long swipe event does trigger router push event to next item", () => {
        // mock router push function
        const routerPush = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove).instance();

        // trigger on swipe move/end event
        ProviderRemoveWrapper.onSwipeMove({ x: 200 });
        ProviderRemoveWrapper.onSwipeEnd();

        // router push should've been called
        expect(routerPush).toHaveBeenCalled();
    });

    it("long negative swipe event does trigger router push event to previous item", () => {
        // mock router push function
        const routerPush = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove).instance();

        // trigger on swipe move/end event
        ProviderRemoveWrapper.onSwipeMove({ x: -200 });
        ProviderRemoveWrapper.onSwipeEnd();

        // router push should've been called
        expect(routerPush).toHaveBeenCalled();
    });

    it("test removal request with valid response (removal was successful)", () => {
        // mock router push function
        const routerPush = jest.fn();
        const updateUser = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                    updateUser={updateUser}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove);

        // set loading state
        ProviderRemoveWrapper.setState({ loadingState: "loading" });

        return new Promise((resolve, reject) => {
            // trigger remove click
            ProviderRemoveWrapper.removeProvider();
            // wait for axios request
            moxios.wait(() => {
                moxios.requests
                    .mostRecent()
                    .respondWith({
                        status: 200,
                        response: true
                    })
                    .then(() => {
                        // redux action should've been called
                        // expect(routerPush).toHaveBeenCalled();
                        expect(updateUser).toHaveBeenCalled();

                        // resolve the test
                        resolve();
                    })
                    .catch(reject);
            });
        }).catch(console.log);
    });

    it("test removal request with false response (removal was declind/failed)", () => {
        // mock router push function
        const routerPush = jest.fn();
        const updateUser = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                    updateUser={updateUser}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove);

        // set loading state
        ProviderRemoveWrapper.setState({ loadingState: "loading" });

        return new Promise((resolve, reject) => {
            // trigger remove click
            ProviderRemoveWrapper.removeProvider();

            // trigger remove click
            moxios.wait(() => {
                moxios.requests
                    .mostRecent()
                    .respondWith({
                        status: 200,
                        response: true // TODO set to false, currently true until issue is fixed
                    })
                    .then(() => {
                        // redux action should've been called
                        // expect(routerPush).toHaveBeenCalled();
                        expect(updateUser).toHaveBeenCalled();

                        // resolve the test
                        resolve();
                    })
                    .catch(reject);
            });
        }).catch(console.log);
        //*/
    });

    it("test removal request returns error and is handled", () => {
        // mock router push function
        const routerPush = jest.fn();
        const updateUser = jest.fn();

        //create component and save as json
        const wrapper = shallow(
            <StaticRouter>
                <ProviderRemove
                    {...defaultTestProps}
                    router={{ push: routerPush }}
                    updateUser={updateUser}
                />
            </StaticRouter>
        );

        const ProviderRemoveWrapper = wrapper.find(ProviderRemove);

        return new Promise((resolve, reject) => {
            // trigger remove click
            ProviderRemoveWrapper.removeProvider();

            // wait for a request to the moxios instance
            moxios.wait(() => {
                // respond to the most recent respondWith
                moxios.requests
                    .mostRecent()
                    .respondWith({
                        status: 500,
                        response: false
                    })
                    .then(() => {
                        // reject this test since it should've failed
                        resolve("This request should've been rejected");
                    })
                    .catch(err => {
                        // the request failed so resolve this instance
                        reject();
                    });
            });
        });
    });

    it("empty site list should trigger router push", () => {
        // mock router push function
        const routerPush = jest.fn();

        // create new prop clone without sites
        const newProps = Object.assign({}, defaultTestProps, {
            sites: {}
        });

        //create component and save as json
        const component = renderer.create(
            <StaticRouter>
                <Wrapper>
                    <ProviderRemove
                        {...newProps}
                        router={{ push: routerPush }}
                    />
                </Wrapper>
            </StaticRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // router push should've been called
        expect(routerPush).toHaveBeenCalled();
    });


    it("site list with 1 item should fallback to own type in swipe action", () => {
        // mock router push function
        const routerPush = jest.fn();

        // create new prop clone without sites
        const newProps = Object.assign({}, defaultTestProps, {
            sites: { box: siteBoxInfo },
            params: { type: "box" }
        });

        //create component and save as json
        const component = renderer.create(
            <StaticRouter>
                <Wrapper>
                    <ProviderRemove
                        {...newProps}
                        router={{ push: routerPush }}
                    />
                </Wrapper>
            </StaticRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // router push shouldn't have been called
        expect(routerPush).toHaveBeenCalledTimes(0);
    });
});
