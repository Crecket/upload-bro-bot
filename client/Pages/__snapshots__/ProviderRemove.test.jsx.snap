// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ProviderRemove /> empty site list should trigger router push 1`
] = `null`;

exports[`<ProviderRemove /> matches snapshot 1`] = `
<div
  className="col-xs-12 col-md-8 col-md-offset-2 col-lg-6 col-lg-offset-3"
  style={
    Object {
      "marginTop": 20,
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#353535",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "#ffffff",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "padding": 20,
        "textAlign": "center",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
      }
    }
  >
    <div
      className={undefined}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="row around-xs"
      >
        <div
          className="col-xs-12"
        >
          <h1>
            Remove 
            Google Drive
            ?
          </h1>
          <p>
            This will disable 
            Google Drive
             service for UploadBro
          </p>
        </div>
        <div
          className="col-xs-12"
        >
          <img
            src="/assets/img/google.svg"
            style={
              Object {
                "height": 120,
                "padding": 20,
              }
            }
          />
        </div>
        <div
          className="col-xs-6 col-md-4"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className={undefined}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "backgroundColor": "#353535",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "minWidth": 88,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#303F9F",
                    "border": 10,
                    "borderRadius": 2,
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "height": 36,
                    "lineHeight": "36px",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": null,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": false,
                        "borderRadius": 2,
                        "height": 36,
                        "muiPrepared": true,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                  >
                    Cancel
                  </div>
                </div>
              </button>
            </div>
          </a>
        </div>
        <div
          className="col-xs-6 col-md-4"
        >
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#353535",
                "borderRadius": 2,
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                       0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "#ffffff",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <button
              disabled={false}
              href={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#448AFF",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "backgroundColor": false,
                      "borderRadius": 2,
                      "height": 36,
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  Remove
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[
  `<ProviderRemove /> site list with 1 item should fallback to own type in swipe action 1`
] = `
<div
  className="col-xs-12 col-md-8 col-md-offset-2 col-lg-6 col-lg-offset-3"
  style={
    Object {
      "marginTop": 20,
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#353535",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "#ffffff",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "padding": 20,
        "textAlign": "center",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
      }
    }
  >
    <div
      className={undefined}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="row around-xs"
      >
        <div
          className="col-xs-12"
        >
          <h1>
            Remove 
            Box
            ?
          </h1>
          <p>
            This will disable 
            Box
             service for UploadBro
          </p>
        </div>
        <div
          className="col-xs-12"
        >
          <img
            src="/assets/img/box.svg"
            style={
              Object {
                "height": 120,
                "padding": 20,
              }
            }
          />
        </div>
        <div
          className="col-xs-6 col-md-4"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            <div
              className={undefined}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "backgroundColor": "#353535",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                         0 1px 4px rgba(0, 0, 0, 0.12)",
                  "boxSizing": "border-box",
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "minWidth": 88,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <button
                disabled={false}
                href={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                onTouchTap={[Function]}
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                    "backgroundColor": "#303F9F",
                    "border": 10,
                    "borderRadius": 2,
                    "boxSizing": "border-box",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontFamily": "Roboto, sans-serif",
                    "fontSize": "inherit",
                    "fontWeight": "inherit",
                    "height": 36,
                    "lineHeight": "36px",
                    "margin": 0,
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "verticalAlign": null,
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <div
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": false,
                        "borderRadius": 2,
                        "height": 36,
                        "muiPrepared": true,
                        "top": 0,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                  >
                    Cancel
                  </div>
                </div>
              </button>
            </div>
          </a>
        </div>
        <div
          className="col-xs-6 col-md-4"
        >
          <div
            className={undefined}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#353535",
                "borderRadius": 2,
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                       0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "#ffffff",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <button
              disabled={false}
              href={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#448AFF",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
              type="button"
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <div
                  style={
                    Object {
                      "backgroundColor": false,
                      "borderRadius": 2,
                      "height": 36,
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  Remove
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
